### PR TITLE
Fix failed webhook delivery preventing it from being delivered again

### LINKED ISSUE #210: When a failed webhook delivery prevents it from being delivered again (with different args).
### Reproduction Steps

+ Create a webhook
+ trigger it and verify it works fine (e.g. check logs)
+ now simulate a failure during the delivery, e.g. a fatal here: 
https://github.com/gocodebox/lifterlms-rest/blob/1.0.0-beta.16/includes/models/class-llms-rest-webhook.php#L32
(basically after that the `pending_delivery` field has been set to 1)
or just set the value directly in the db :)
+ trigger the same webhook

### Expected Behavior

+ I would expect the second webhook to be delivered

### Actual Behavior

+ The second webhook is not delivered.
This is because the `pending_delivery` is set to 1 in the db for that webhook, and then `should_deliver()` returns false:
https://github.com/gocodebox/lifterlms-rest/blob/1.0.0-beta.16/includes/models/class-llms-rest-webhook.php#L421

### Error Messages / Logs

n/a

### System Information

<details>
<summary>System Report</summary>
  
```
LifterLMS Rest 1.0.0-beta.16

```

</details>

### Browser, Device, and Operating System Information

n/a

### Related User Information
HS-140845

Solution here would be to not store the `pending_delivery` and, of course, not relying on this.
I've investigated a little bit into this and IF (as it seems) the reason why we use it is to avoid delivery duplicate within the same request, then we should use a different approach, e.g. looking at our inspiration:
https://github.com/woocommerce/woocommerce/commit/b262ac8e53fe20fbe97bb65332fabaa0b0752c1c#diff-8971231b02d400aadd63a60794ca17128d5dd8e9ed670745f6785ec8d9425128

Additionally, the same inspiring software, uses the `pending_delivery` **only** for deciding whether or not a delivery ping should be performed: sends the delivery ping only if `pending_delivery` is set to 1 (and this value is set on the webhook creation).

We use a different approach to determine it, and already don't rely on the `pending_delivery` property.
(I think we could even get rid of it...)

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
## Description
Fixes #210

## How has this been tested?
Manually and with existing and new tests.

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards. 

@thomasplevy 
as said in #210 
we could even totally remove the `pending_delivery` property and the logic related to it. I didn't do it yet because I wanted to know your opinion on this. I guess that, since we're in beta, we can get rid of public methods without problems, right?

Also, we should remove the related db column, this means that we need an update routine. But you know, I don't know if it's worthwhile to implement all the "bg db update" logic here, or just implement this minor update in core or whatever. Ideas?

### CODE REVIEW

File: includes/models/class-llms-rest-webhook.php [hunk 1]
```diff
@@ -472,11 +499,10 @@ protected function schedule( $args ) {
 		 * @param array             $args  Numeric array of arguments from the originating hook.
 		 * @param LLMS_REST_Webhook $this  Webhook object.
 		 */
-		$delay = apply_filters( 'llms_rest_webhook_repeat_delay', 300, $args, $this );
+		$delay = apply_filters( 'llms_rest_webhook_repeat_delay', 3, $args, $this );
```
Comment (thomasplevy) [inline includes/models/class-llms-rest-webhook.php]: Is this an intentional change or was it accidentally committed after debugging?

**Review by thomasplevy — COMMENTED** [review]
I think you mentioned somewhere that we basically don't need the `pending_delivery` record in the DB. After reading this it seems that it's useless now, right? 

NOTE: I didn't review all the code in the codebase I'm working quickly off my (very poor) memory.

Is this something we should remove while issuing this fix? Or am I misreading/remembering?

File: includes/models/class-llms-rest-webhook.php (hunk 1 above)
Comment (eri-trabiccolo) [reply includes/models/class-llms-rest-webhook.php]: shite

Comment (eri-trabiccolo) [reply includes/models/class-llms-rest-webhook.php]: fixed in  3775efa

**Comment by eri-trabiccolo** [comment]
> I think you mentioned somewhere that we basically don't need the `pending_delivery` record in the DB. After reading this it seems that it's useless now, right?

It is useless, that column now is useless, see my comment below the PR description (I will put it at the end of this reply as well).

> NOTE: I didn't review all the code in the codebase I'm working quickly off my (very poor) memory.
> 
> Is this something we should remove while issuing this fix? Or am I misreading/remembering?

You remember well.

----
@thomasplevy
as said in #210
we could even totally remove the pending_delivery property and the logic related to it. I didn't do it yet because I wanted to know your opinion on this. I guess that, since we're in beta, we can get rid of public methods without problems, right?

Also, we should remove the related db column, this means that we need an update routine. But you know, I don't know if it's worthwhile to implement all the "bg db update" logic here, or just implement this minor update in core or whatever. Ideas?

**Comment by thomasplevy** [comment]
I obviously read the PR message and didn't just jump right into the code review... 

Anyway,

> we could even totally remove the pending_delivery property and the logic related to it. I didn't do it yet because I wanted to know your opinion on this. I guess that, since we're in beta, we can get rid of public methods without problems, right?

Yea, it's "safe" enough since it's beta and the change is recorded in the changelog clearly.

> Also, we should remove the related db column, this means that we need an update routine. But you know, I don't know if it's worthwhile to implement all the "bg db update" logic here, or just implement this minor update in core or whatever. Ideas?

I don't truly know either. I suppose we can just leave a column unused and hanging but that feels messy to me and I'd rather clean it up.

Although it is okay with me if you'd rather split that work into a separate issue that can be worked on later. If you'd rather do that please record the issue so we don't forget about it!

**Comment by eri-trabiccolo** [comment]
@thomasplevy 
Opened an issue to track the fact that we have to remove that column from the db (old installs).

File: includes/class-llms-rest-webhooks.php [hunk 2]
```diff
@@ -77,9 +78,6 @@ public function create( $data ) {
 			return $data;
 		}
 
-		// Can't set these properties during creation.
-		unset( $data['pending_delivery'], $data['failure_count'] );
```
Comment (thomasplevy) [inline includes/class-llms-rest-webhooks.php:81]: We should be retaining the `failure_count` column though, right?

File: includes/class-llms-rest-webhooks.php [hunk 3]
```diff
@@ -552,7 +551,6 @@ protected function update_prepare( $data ) {
 
 		// Merge in (some) default values.
 		$defaults = $this->get_default_column_values();
-		unset( $defaults['pending_delivery'], $defaults['failure_count'] );
```
Comment (thomasplevy) [inline includes/class-llms-rest-webhooks.php:555]: `failure_count` should stay?

**Review by thomasplevy — CHANGES_REQUESTED** [review]
Looks good except what appears to be the accidental deletion of the unsetting of `failure_count`.

File: includes/class-llms-rest-webhooks.php (hunk 3 above)
Comment (eri-trabiccolo) [reply includes/class-llms-rest-webhooks.php:555]: yes, sorry

**Comment by thomasplevy** [comment]
OH NO CS :-(

**Pull request merged**